platforms/gke/base/_shared_config/configmanagement_variables.tf

```diff
@@ -26,6 +29,8 @@ locals {
   oci_root_sync_image_tag  = "latest"
   oci_sync_repo            = var.configmanagement_sync_repo == null ? local.oci_sync_repo_url : var.configmanagement_sync_repo
   oci_sync_repo_url        = "${local.oci_repo_url}/${local.oci_root_sync_image}"
+
+  wi_principal_prefix = "principal://iam.googleapis.com/projects/${data.google_project.cluster.number}/locations/global/workloadIdentityPools/${data.google_project.cluster.project_id}.svc.id.goog/subject"
```

Comment:
This creates a dependency on a resource, so any other service that tries to use this shared config would require the same resource. Is there a way we can do this with variables or move it into the service?

Reply:
I think we can just move it in a `local` var. Would that be ok?

Reply:
It would be nice to have it defined here or in the cluster variables file so we can reuse it. I think it's something other services might want to use.

Reply:
I suppose we could get around the project ID by using the variable directly, but I don't think we have the project number readily available as a variable.

Reply:
Yes, so any "derived" values need to exist in the actual service Terraform. Otherwise we need to include them as a "defined" variable in a `.tfvars` file. 

So then it becomes, is there enough benefit in having them specify the value or should we just pull it where we need it.

Reply:
I moved the `wi_principal_prefix` variable definition in `iam.tf` for both the `oci` and the `git` modules. I also refactored the definition of `oci_repo_url` to use `var.cluster_project_id` to remove a dependency on a data resource in that case as well.

Reply:
PTAL, thanks.